src/classes/interactionpotential.h

```diff
@@ -32,7 +32,17 @@ template <class Functions> class InteractionPotential
 
     public:
     // Set functional form of interaction
-    void setForm(typename Functions::Form form) { form_ = form; }
+    void setForm(typename Functions::Form form)
+    {
+        form_ = form;
+        // Set parameters vector to correct size
+        while (parameters_.size() != Functions::forms().minArgs(form_))
+            if (parameters_.size() < Functions::forms().minArgs(form_))
```

Comment:
Ah! I think you can blame me for this one. The logic here is sound, but the use of the `minArgs()` function is the problem.  If you look in `src/base/enumoptions.h` on line 169 where this function is defined you'll see that it actually returns `std::optional<int>` rather than just an `int`.  So, if no minimum number of arguments is defined in the option, it returns `std::nullopt` (i.e. no `int` value at all) rather than zero (which is a specific minimum value).

So, our check on the number of parameters can actually end up comparing `parameters_size()` with `std::nullopt`, which doesn't end well and causes either an infinite `pop_back()` or an infinite `push_back()` loop.  Neither are good!

The solution is to use a default value (zero) when no minimum value is defined:
```suggestion
        while (parameters_.size() != Functions::forms().minArgs(form_).value_or(0))
            if (parameters_.size() < Functions::forms().minArgs(form_).value_or(0))
```